src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -52,46 +53,64 @@ public partial class NodeViewModel : ViewModelBase
 
         #region public members
 
+        /// <summary>
+        /// Returns NodeModel ID
+        /// </summary>
+        public Guid Id
```

Comment:
It's a read-only property (no set function) so it shouldn't be involved in deserialization.